src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/EmploymentCheck/RefreshEmploymentChecksCommandHandler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Commands.Types.ApprenticeshipIncentive;
+using SFA.DAS.EmployerIncentives.Domain.Interfaces;
+
+namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.EmploymentCheck
+{
+    public class RefreshEmploymentChecksCommandHandler : ICommandHandler<RefreshEmploymentChecksCommand>
+    {
+        private readonly IApprenticeshipIncentiveDomainRepository _incentiveDomainRepository;
+        private readonly ICommandPublisher _commandPublisher;
+
+        public RefreshEmploymentChecksCommandHandler(IApprenticeshipIncentiveDomainRepository incentiveDomainRepository, 
+                                                     ICommandPublisher commandPublisher)
+        {
+            _incentiveDomainRepository = incentiveDomainRepository;
+            _commandPublisher = commandPublisher;
+        }
+        public async Task Handle(RefreshEmploymentChecksCommand command, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var incentivesWithLearning = await _incentiveDomainRepository.FindIncentivesWithLearningFound();
+
+            foreach(var incentive in incentivesWithLearning)
+            {
+                if (!incentive.HasEmploymentChecks)
+                {
+                    incentive.AddEmploymentChecks();
+                    await _incentiveDomainRepository.Save(incentive);
+                }
+
+                await _commandPublisher.Publish(new SendEmploymentCheckRequestsCommand(incentive.Id));
```

Comment:
When AddEmploymentChecks is called the EmploymentChecksCreated event is raised.  The EmploymentChecksCreatedHandler already handles that event and publishes the SendEmploymentCheckRequestsCommand.  So this line can be removed completely.

Reply:
I think it's still needed for when the incentive already has employment checks that need refreshing, so will add an else condition

Reply:
But that AddEmploymentChecks method clears them and replaces them doesn't it ?

```
 public void AddEmploymentChecks()
        {
            Model.EmploymentCheckModels.Clear();
```

Reply:
So use AddEmploymentChecks regardless of whether the incentive already has checks.

Reply:
Isn't the use case to **force** a **new** employment check for all apprenticeships that have a learner record and haven't been withdrawn?  Double check with Mike but I think existing checks are to be replaced.  This may be required if the rules change and all existing incentives have to be re-checked.

Reply:
Yep, they all need to be rechecked for this process.